src/utransport.rs

```diff
@@ -25,6 +26,41 @@ use zenoh::{
     queryable::Query,
 };
 
+#[inline]
+fn error_callback(
+    sender: &Sender<CallbackChannelMessage>,
+    listener: &Arc<dyn UListener>,
+    msg: &str,
+) {
+    log::error!("{msg}");
+    if sender
+        .send(CallbackChannelMessage {
+            listener: listener.clone(),
```

Comment:
Some advice I read to make APIs more explanatory is that if we're always going to clone an Arc inside a function, we might as well ask for `Arc<T>` instead of `&Arc<T>`.

I'm sure it doesn't fit everywhere, but I'd be interested to hear your thoughts.

Reply:
This is what clippy asks me to do :laughing: 

```
error: this argument is passed by value, but not consumed in the function body
  --> src/utransport.rs:50:15
   |
50 |     listener: Arc<dyn UListener>,
   |               ^^^^^^^^^^^^^^^^^^ help: consider taking a reference instead: `&Arc<dyn UListener>`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_pass_by_value
```

I guess the clippy doesn't think it necessary to bump up /down the reference count here.
Here is the explanation from others:
https://stackoverflow.com/questions/55576425/why-does-clippy-suggests-passing-an-arc-as-a-reference